File: Basic/Calculate Factorial of A Number/SolutionByRishabh.java
/*Program to calculate factorial of a number*/

import java.util.Scanner;

public class Factorial {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in); //Initialising object of Scanner Class
        System.out.println("Enter the number"); // Message to inform user what to do next
        int n = sc.nextInt(); //Assigning user input to variable n
        int fact = 1; //Assigning 1 to fact
        //Process to calculate factorial
        for (int i = 1; i <= n; i++) {
            fact = i * fact;
        }
        System.out.println("Output = " + fact); //Printing the result
    }
}

File: Basic/Calculate Factorial of A Number/SolutionByRishabh.cpp
#include <iostream>
using namespace std;

int fact(int n)
{
	if(n==1)
		return 1;
	else
		return n*fact(n-1);
}
int main()
{
	int n;
	cout<<"\n Enter the Number ";
	cin>>n;

	cout<<fact(n);
}
